File: API_SC_01_08Abril/Lincoln Matheus/API_LINCOLN_MATHEUS/API_SC_2024.1/q4_viagem.js
import { get_number, get_text, porcentagem, print } from "../Utils/utils.js";

function main() {
    const ORIGEM = get_text(`Digite de onde voce esta partindo: `).toUpperCase();
    const DESTINO = get_text(`Digite para onde voce deseja ir: `).toUpperCase();
    const VALOR_REAIS = get_number(`Digite quanto custa(R$) o valor da viagem no site: `);
    const VALOR_MILHAS_SITE = get_number(`Digite o valor em milhas do site: `);
    const VALOR_MILHEIRO_PADRAO = 0.07;
    const VALOR_MILHEIRO_BARATO = 0.0145;

    let valorMilhasPadraoEmReal = VALOR_MILHAS_SITE * VALOR_MILHEIRO_PADRAO;
    let valorMilhasBaratasEmReal =  VALOR_MILHAS_SITE * VALOR_MILHEIRO_BARATO;

    let percentualMilhasPadrao = porcentagem(valorMilhasPadraoEmReal, VALOR_REAIS);
    let percentualMilhasBaratas = porcentagem(valorMilhasBaratasEmReal, VALOR_REAIS);

    let melhorEscolha = verificarMelhorEscolha(VALOR_REAIS, valorMilhasPadraoEmReal, valorMilhasBaratasEmReal);

    print(`
    ----------------------------------------------------------------------------
    >>>>>>>>>>>>>>>>                MELHOR ESCOLHA              >>>>>>>>>>>>>>>>
    
    -> Origem: -----------------------------------------           ${ORIGEM}
    -> Destino: ----------------------------------------           ${DESTINO}
    -> Valor em Reais   --------------------------------           R$:${VALOR_REAIS.toFixed(2)}
    -> Valor em Milhas  --------------------------------           ${VALOR_MILHAS_SITE}

    -> Valor em Milhas Padrão(R$)   --------------------           R$:${valorMilhasPadraoEmReal.toFixed(2)}
    -> Valor em Milhas Baratas(R$)  --------------------           R$:${valorMilhasBaratasEmReal.toFixed(2)}
    -> Percentual de Milhas P. em relação a Valor em R$            ${percentualMilhasPadrao.toFixed(2)}%
    -> Percentual de Milhas B. em relação a Valor em R$            ${percentualMilhasBaratas.toFixed(2)}%
    
    ___________________________________________________________________________
    
    Logo, podemos concluir que a Melhor Escolha é:  ---            ${melhorEscolha}!\n`);


}

function verificarMelhorEscolha(opcao1, opcao2, opcao3) {
    let melhor = ``;

    if(opcao1 < opcao2 && opcao1 < opcao3) {
        melhor = `Valor em Reais(R$)`;
    } else if(opcao2 < opcao1 && opcao2 < opcao3) {
        melhor = `Valor em Milhas Padrão`;
    } else if (opcao3 < opcao1 && opcao3 < opcao2) {
        melhor = `Valor em Milhas Baratas`;
    }

    return melhor;
}

main();

File: API_SC_01_08Abril/Thalysson Delano/apisc.py/q4_viagem.py
def main():
    
    print('Plano de Viagem\n--------------')

    origem = str(input('Digite a Origem: '))
    destino = str(input('Digite o Destino: '))
    valor_reais = float(input('Digite o Valor em Reais : '))
    valor_milhas = float(input('Digite o Valor em Milhas '))

    passagem_milhas_padrao = (valor_milhas / 1000) * 70
    passagem_milhas_baratas = (valor_milhas / 1000) * 14.50
    porcentagem_milhas_baratas_reais = (passagem_milhas_baratas / valor_reais) * 100
    porcentagem_milhas_padrao_reais = (passagem_milhas_padrao / valor_reais) * 100

    print(f'''   ======== RELATORIO ========
    Valor em Reais com Milhas Padrão : R${passagem_milhas_padrao:.2f}
    Porcentagem em Relação ao Valor em Reais ( Dinheiro ) : {porcentagem_milhas_padrao_reais:.1f}%

    Valor em Reais com Milhas Baratas: R${passagem_milhas_baratas:.2f}
    Porcentagem em Relação ao Valor em Reais ( Dinheiro ) : {porcentagem_milhas_baratas_reais:.1f}%
    _____________________________________________________________________''')

    if valor_reais < passagem_milhas_baratas and valor_reais < passagem_milhas_padrao:
        print('A OPÇÃO MAIS VANTAJOSA É COMPRAR EM DINHEIRO')
    elif passagem_milhas_baratas < valor_reais and passagem_milhas_baratas < passagem_milhas_padrao:
        print('A OPÇÃO MAIS VANTAJOSA É COMPRAR USANDO MILHAS BARATAS')
    else:
        print('A OPÇÃO MAIS VANTAJOSA É COMPRAR USANDO MILHAS PADRÃO')

main()